### ISSUE TITLE
Don't hardcode firmware version and build date

### ISSUE THREAD

**Issue by filipleple** [Dasharo/open-source-firmware-validation#632]
### The problem you're addressing (if any)

Right now, we're hardcoding the build date and firmware version into each platform config in the 
`${DMIDECODE_FIRMWARE_VERSION}` and `${DMIDECODE_RELEASE_DATE}` flags. This means we have to do trivial edits of the configs for each release/RC.

### Describe the solution you'd like

We should come up with a way to do this automatically. I think most if not all could be extracted from the FW file we pass into OSFV. For example:

```
λ strings protectli_v1610_v0.9.3.rom | grep DMI_DATE
#define COREBOOT_DMI_DATE "09/06/2024"
λ strings protectli_v1610_v0.9.3.rom | grep EXTRA_VERSION
#define COREBOOT_EXTRA_VERSION "-v0.9.3"
```
to set the date and version, and 

```
λ strings protectli_v1610_v0.9.3.rom | grep -i uefi
EFIVARS: UEFI FV with size %lld found
EFIVARS: UEFI variable store with size %zu found

λ strings optiplex_9010_v0.1.0-rc1_seabios.rom | grep -i uefi
λ
```

to see if we're dealing with `Dasharo (coreboot + EDK2)` or `Dasharo (coreboot + SeaBIOS)`

### Where is the value to a user, and who might that user be?

The testers wouldn't have to make trivial changes to the configs each release and PRs with such changes wouldn't have to be handled

### Describe alternatives you've considered

-

### Additional context

-

**Comment by philipanda** [comment]
Won't that make the test always pass? Isn't a part of the DMI test suite to check if the values, like the version of the firmware, are as expected?
Reading the valid values from the fw image would essentially check if the values are properly read by dmidecode, not if they are valid. 

If we could automate the process of acquiring the binary, the FW version, build time etc. could be compared to the actual expected values, not what is currently in the FW https://github.com/Dasharo/open-source-firmware-validation/issues/662#issuecomment-2601489553

**Comment by miczyg1** [comment]
There are two aspects of testing DMI:

1. Whether the values exposed in dmidecode match what is in the binary.
2. Whether the dmidecode values match what is expected to be.

There is no other way  to test 2nd case than hardcoding the version or passing it as parameter. Ideally we would match all 3 values:
1. The passed parameter (expected version).
2. The version in the binary
3. The version displayed in the dmidecode output.

**Comment by philipanda** [comment]
> There is no other way to test 2nd case than hardcoding the version or passing it as parameter. 

Exactly, it can be passed as a parameter. But the tester doesn't need to write it down manually.
If we could automatically download the binary from CI from a given tag, instead of passing the file manually, the version could be determined from the tag name. Maybe the build time too?
